src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -197,8 +197,10 @@ private static SyntaxNode WithBlockBody(SyntaxNode node, BlockSyntax body)
         {
             declarationStatement = declarationStatement.WithAdditionalAnnotations(Formatter.Annotation);
 
-            var oldOutermostBlock = block;
-            var matches = FindMatches(document, expression, document, oldOutermostBlock, allOccurrences, cancellationToken);
+            // If we're within a local function, our scope for the new local declaration is expanded to include the enclosing member.
+            var scope = block.GetAncestor<LocalFunctionStatementSyntax>() == null ? block : (SyntaxNode)block.GetAncestor<MemberDeclarationSyntax>();
```

Comment:
I'm not familiar with the rest of the service, but it seems like from here this will expand the scope for static local functions as well?  But I think it shouldn't for static local since the var inside the static local function isn't visible.

If it's already covered somewhere else, might be good to add a test.

Reply:
I think you're correct, thanks for the catch! I updated the code with additional tests.